src/pip/_internal/vcs/git.py

```diff
@@ -259,12 +259,18 @@ def fetch_new(self, dest, url, rev_options):
         # type: (str, HiddenText, RevOptions) -> None
         rev_display = rev_options.to_display()
         logger.info('Cloning %s%s to %s', url, rev_display, display_path(dest))
-        self.run_command(make_command('clone', '-q', url, dest))
+        if self.get_git_version() >= (2, 17):
+            self.run_command(make_command(
+                'clone', '--filter=blob:none', '-q', url, dest,
+            ))
+        else:
+            self.run_command(make_command('clone', '-q', url, dest))
```

Comment:
Should add comment to describe what we’ve discussed above so future readers know the context.